conans/client/installer.py

```diff
@@ -419,11 +418,13 @@ def _build(self, nodes_by_level, keep_build, root_node, graph_info, remotes, bui
                     assert ref.revision is not None, "Installer should receive RREV always"
                     if node.binary == BINARY_UNKNOWN:
                         self._binaries_analyzer.reevaluate_node(node, remotes, build_mode, update)
+                        if node.binary == BINARY_MISSING:
```

Comment:
This is the key to not checking the ``package_folder exists``. If after re-evaluation the node is missing, we need to raise early, otherwise the later ``call_package_info()`` in the ``_handle_node_cache`` will raise, as the package folder does not exist.